### DISCUSSION TITLE
Need for Better Communication from Maintainers

### DISCUSSION THREAD

**Discussion by Nantris** [ueberdosis/tiptap#5167] [General Questions] [answered]
I've noticed that regressions are being introduced and not addressed in a timely fashion. Additionally, sometimes issues are closed without being fixed. This is not to say these problems affect all issues, but they do affect some important ones.

I try to have reasonable expectations for open source and to help where I can, but there's no replacement for communication from the maintainers. I'm not asking for updates on every issue, but especially on regressions and where maintainers have made promises or assurances, I would expect to see some progress or even just updates. It's becoming hard to remain hopeful when there's so little communication from the team.

One example is this bug introduced in 2.2.0: [PR #5060](https://github.com/ueberdosis/tiptap/pull/5060). I submitted a simple fix for it 40 days ago: [PR #5061](https://github.com/ueberdosis/tiptap/pull/5061). Since then, two minor versions have been released, but the regression remains unaddressed. This issue was first present as far back as November 2023 and the biggest reason I couldn't spot it until more recently is due to other unaddressed bugs (eg #4704 #4862) which prevent our upgrade path even now nearly 4 months after first reporting it. This is really discouraging and is increasingly compounding upon itself at this point.

**Some suggestions to improve the situation:**

1. **Better Communication:** Regular updates on critical issues, regressions, and promises.
2. **Issue Management:** Prioritize and address regressions promptly. Fixes for regressions should not be considered breaking changes - it is the regressions that are the breaking changes. The failure to promptly address regressions is what's turning their fixes into breaking changes.
3. **Acknowledgment of Contributions:** Timely review and merge of community fixes. People will be encouraged to contribute if their hard work makes a difference, and discouraged if it does not. Good contributions can become outdated, which can be discouraging for contributors.

I hope this feedback is received in the constructive spirit in which it's intended. Great work has been done on TipTap. It's a remarkable effort to make ProseMirror more accessible, and even more powerful. Still, I do feel the need to raise these issues as they're increasingly causing problems for us.

Thank you for considering these suggestions.

**Comment by nperez0111** [comment, marked as answer]
Appreciate your suggestions here.

So you are aware, we are spending a bunch of time building our own products and it is difficult to make prioritization calls between projects. I've recently joined and have been ramping up and making my own contributions and have a lot of excitement in trying to do open source "right" but it can be overwhelming at times with the number of ways that people submit issues (github issues, PRs, discord, slack).

We are trying to find that balance, and I've personally tried to resolve some of the issues that you have mentioned.

**Reply by Nantris** [reply]
@nperez0111 I appreciate your reply and your efforts have not gone unnoticed! Thank you very much for taking the time to look into the mentioned issues. I certainly understand it's difficult being torn in so many directions at once.

PS: There's a Slack channel in addition to the Discord? It sounds tough indeed to stay on top of all of that. Is there a reason to maintain both channels? The fragmentation between Discord and Github already seems a pervasive problem in open source and so I can't imagine having even a third place to try to manage.

**Reply by nperez0111** [reply]
Yea, fragmentation is real. Some of our paying customers (especially bigger enterprises) have direct access to our Slack.

And good, I hope I'm making an impact around here

**Reply by Nantris** [reply]
Ah that's painful to need it for paying customers! Thank you for your hard work @nperez0111!
